src/systems/filecoin_vm/interpreter/vm_interpreter.id

```diff
@@ -17,6 +17,11 @@ type TipSetMessages struct {
     Epoch   UInt64  // The chain epoch of the blocks
 }
 
+type MessageRef struct {
+    Message          msg.UnsignedMessage
+    MessageSizeOrig  int
+}
+
```

Comment:
See my comment in vm_interpreter.go.

You could replace the array elements in BlockMessages.Messages with this, but it's probably better to represent the BLS and SECP messages explicitly, so this spec can also describe which order they are evaluated in.

Reply:
(deferring pending #669)